tests/src/test/java/alluxio/job/replicate/ReplicateIntegrationTest.java

```diff
@@ -76,4 +99,28 @@ public void replicateLastBlockFromUFS() throws Exception {
     Assert.assertEquals(TEST_BLOCK_SIZE, blockInfo1.getLength());
     Assert.assertEquals(1, blockInfo2.getLength());
   }
+
+  // Tests that we send one request and log one message when job master is at capacity
+  @Test
+  @LocalAlluxioClusterResource.Config(confParams = {PropertyKey.Name.JOB_MASTER_JOB_CAPACITY, "0",
+      PropertyKey.Name.JOB_MASTER_FINISHED_JOB_RETENTION_MS, "0"})
+  public void requestBackoffTest() throws Exception {
+    String rootDir = "/backofftest";
+    for (int i = 0; i < 10; i++) {
+      AlluxioURI uri = new AlluxioURI(rootDir + "/" + i);
+      createFileOutsideOfAlluxio(uri);
+    }
+    String exceptionMsg = ExceptionMessage.JOB_MASTER_FULL_CAPACITY
+        .getMessage(Configuration.get(PropertyKey.JOB_MASTER_JOB_CAPACITY));
+    String replicationCheckerMsg = "The job service is busy, will retry later. " + exceptionMsg;
+    String rpcUtilsMsg = "alluxio.exception.status.ResourceExhaustedException: " + exceptionMsg;
+    SetAttributeOptions opts = SetAttributeOptions.defaults();
+    opts.setReplicationMin(2);
```

Comment:
should this be set on all 10 files?

Reply:
Maybe I'm misunderstanding how the setAttribute works because of the way it works in the CLI but when I run SetReplication on a directory I see the replication propagate to every file. I assumed I could do it the same here.

Reply:
I debugged and confirm that setting replication on the root directory causes all of the underlying files to have their replication's updated. So I think this code is fine.

Reply:
ah, didn't realize it worked recursively by default, sorry about that